File: P5 2024-2/Paradigmas de Programacao/ListaExercicio01/Golang/Calculadora.go
package main

import "fmt"

func somaDois(a float64, b float64) float64 {
	resultado := a + b
	fmt.Println("Resultado da soma:", resultado)
	return resultado
}

func somaTres(a float64, b float64, c float64) float64 {
	resultado := a + b + c
	fmt.Println("Resultado da soma:", resultado)
	return resultado
}

type Produto struct {
	nome  string
	valor float64
}

func somaProdutos(p1 *Produto, p2 *Produto) float64 {
	return somaDois(p1.valor, p2.valor)
}

func fatorial(n int) int {
	resultado := 1
	if n > 0 {
		for i := 1; i <= n; i++ {
			resultado *= i
		}
	} else if n == 0 {
		resultado = 1
	} else {
		resultado = 0 // erro
	}
	fmt.Printf("%d! = %d\n", n, resultado)
	return resultado
}

File: P5 2024-2/Paradigmas de Programacao/ListaExercicio01/Python/Calculadora.py
class Calculadora:
    def __init__(self):
        pass
    
    def soma(self, n1, n2, n3 = 0):
        return n1+n2+n3
    
    @staticmethod
    def fatorial(n):
        if n == 0 or n == 1:
            return 1
        else:
            return n * Calculadora.fatorial(n - 1)
    
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    def __add__(self, produto2):
        return self.preco + produto2.preco